upstream-sha256: 036b36917bdfc59d742cfad7575fa5c428b7bd5ba526c8442a09faeb2f45113d
actual-sha256: 3fac67b01fe836e30cdf2d30a062c79199980051a70aae93523c2791eee62db4
--- upstream.expected
+++ tolk.actual
@@ -18,7 +18,7 @@
   float val4 = (*(data4_16+(lidx1+4)));
   float val5 = (*(data4_16+(lidx1+8)));
   float4 val6 = (*((float4*)((data3_16+alu0))));
-  *(data0_16+(lidx1+alu0)) = ((*(buf0+0))+(val6.x*val2)+(val6.y*val4)+(val6.z*val5)+(val6.w*val3));
+  *(data0_16+(lidx1+alu0)) = ((*(buf0+0))+(0.0f+(val6.x*val2)+(val6.y*val4)+(val6.z*val5)+(val6.w*val3)));
 }
 ---
 typedef unsigned int uint;
@@ -33,7 +33,7 @@
   __shared__ __align__(16) float buf4[16];
   int lidx0 = threadIdx.x; /* 16 */
   float val0 = (*(data1_16+lidx0));
-  *(buf4+lidx0) = (val0*val0);
+  *(buf4+lidx0) = (0.0f+(val0*val0));
   __syncthreads();
   *(buf0+0) = 0.0f;
   for (int Ridx106 = 0; Ridx106 < 16; Ridx106++) {
